lib/methwhy.g

```diff
@@ -388,6 +388,49 @@ BIND_GLOBAL("ShowImpliedFilters",function(filter)
   fi;
 end);
 
+
+#############################################################################
+##
+#F  ShowDeclarationsOfOperation( <oper> )
+##
+##  <#GAPDoc Label="ShowDeclarationsOfOperation">
+##  <ManSection>
+##  <Func Name="ShowDeclarationsOfOperation" Arg='oper'/>
+##
+##  <Description>
+##  Displays information about all declarations of the operation <A>oper</A>,
+##  including the location of each declaration and the argument filters.
+##  <Log><![CDATA[
+##  gap> ShowDeclarationsOfOperation(IsFinite);
+##  Available declarations for operation <Property "IsFinite">:
+##    1: GAPROOT/lib/coll.gd:1451 with 1 arguments, and filters [ IsListOrCollection ]
+##    2: GAPROOT/lib/float.gd:212 with 1 arguments, and filters [ IsFloat ]
+##    3: GAPROOT/lib/ctbl.gd:1195 with 1 arguments, and filters [ IsNearlyCharacterTable ]
+##  ]]></Log>
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+BIND_GLOBAL("ShowDeclarationsOfOperation",function(oper)
+    local locs, reqs, i, r;
+    if not IsOperation(oper) then
+        Error("<oper> must be an operation");
+    fi;
+    Print("Available declarations for operation ", oper, ":\n");
+    locs := GET_DECLARATION_LOCATIONS(oper);
+    reqs := GET_OPER_FLAGS(oper);
+    for i in [1.. Length(locs)] do
+       r := List(reqs[i], r -> JoinStringsWithSeparator(NamesFilter(r), " and "));
+       Print(String(i, 3), ": ", locs[i][1], ":", locs[i][2],
+             " with ", Length(reqs[i]),
+             " arguments, and filters [ ",
+             JoinStringsWithSeparator(r, ", "),
+             " ]\n"
```

Comment:
Because of the way the strings are joined together, when a line in the output is too long it breaks in an unintuitive way that is hard to read.
Therefore I suggest first build a complete string and then print it (or something similar).
An example where this happens is by invoking
`ShowDeclarationsOfOperation(IsPrimitive);`
The result is (for example depending on the width of the terminal):
```Available declarations for operation <Property "IsPrimitive">:
  1: /mnt/c/gap/lib/oprt.gd:1879 with 1 arguments, and filters [ IsExternalSet ]
  2: /mnt/c/gap/lib/oprt.gd:1879 with
5
  arguments, and filters [ IsMagmaWithInverses and IsAssociative and HasIsAssociative, IsListOrCollect\
ion, IsList, IsList, IsFunction ]```

For instance something like the following might work:
```Print(JoinStringsWithSeparator([String(i, 3), ": ", locs[i][1], ":", locs[i][2],
             " with ", Length(reqs[i]),
             " arguments, and filters [ ",
             JoinStringsWithSeparator(r, ", "),
             " ]\n"],"")
             );```

Reply:
I've tweaked the output now by adding some `\c` into the string being printed, so now one gets this in your example:
```
gap> SizeScreen([80,100]);
[ 80, 100 ]
gap> ShowDeclarationsOfOperation(IsPrimitive);
Available declarations for operation <Property "IsPrimitive">:
  1: /Users/mhorn/Projekte/GAP/gap.spielwiese/lib/oprt.gd:1831 with 1 arguments, and filters [ IsExternalSet ]
  2: /Users/mhorn/Projekte/GAP/gap.spielwiese/lib/oprt.gd:1831 with 5 arguments, and filters [ IsMagmaWithInverses and IsAssociative and HasIsAssociative, IsListOrCollection\
, IsList, IsList, IsFunction ]]
```

Reply:
Anyway, this is a debugging tool; while I am willing to make trivial improvements, there is only so far I'd want to with polishing it, as the returns are quickly diminishing.